cloud-manager/pkg/providers/aws_provider.go

```diff
@@ -0,0 +1,108 @@
+package providers
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/pkg/errors"
+)
+
+type awsProvider struct {
+	Provider
+	name      string
+	profile   string
+	region    string
+	ec2Client ec2iface.EC2API
+}
+
+func NewAwsProvider(name, profile, region string, ec2Service ec2iface.EC2API) (Provider, error) {
+	if name == "" || profile == "" || region == "" {
+		return nil, errors.New("one or all required attributes(name, profile, region) is blank")
+	}
+
+	ec2Svc := ec2Service
+
+	if ec2Svc == nil {
+		authSession, err := newSession(profile, region)
+
+		if err != nil {
+			return nil, err
+		}
+
+		ec2Svc = ec2.New(authSession)
+	}
+
+	return &awsProvider{
+		name:      name,
+		profile:   profile,
+		region:    region,
+		ec2Client: ec2Svc,
+	}, nil
+}
+
+func newSession(profile, region string) (*session.Session, error) {
+
+	config := &aws.Config{
+		Region:      aws.String(region),
+		Credentials: credentials.NewSharedCredentials("", profile),
+	}
+
+	return session.NewSession(config)
+}
+
+func (ap *awsProvider) GetName() string {
+	return ap.name
+}
+
+func (ap *awsProvider) GetInstance(privateDnsName string) (*ec2.Instance, error) {
+	input := &ec2.DescribeInstancesInput{
+		Filters: []*ec2.Filter{
+			{
+				Name:   aws.String("private-dns-name"),
+				Values: aws.StringSlice([]string{privateDnsName}),
+			},
+		},
+	}
+	response, err := ap.ec2Client.DescribeInstances(input)
+	if err != nil {
+		return nil, err
+	}
+	reservations := response.Reservations
+	if len(reservations) == 0 {
+		return nil, nil
+	}
+
+	instances := reservations[0].Instances
+	if len(instances) == 0 {
+		return nil, nil
+	}
+
+	return instances[0], nil
+}
+
+func (ap *awsProvider) TerminateInstance(instanceName string) (bool, error) {
```

Comment:
should we check if the instance is protected for shutdown? and if yes remove the protection?

Reply:
Yes you are right @cpanato, but I believe that If we add the protection removal it will defeat the purpose of it, as someone supposedly has enabled it for a specific reason.
Or what we can do, is to parse it as a new flag, or to enable it alongside with the `force` flag?

Reply:
sounds great!

Reply:
I've added the flag `-force-termination` which disables the termination protection, otherwise it fails and suggests to use that flag if necessary.